src/streaming.rs

```diff
@@ -0,0 +1,270 @@
+use ::open_vcdiff_sys;
+
+use std::error;
+use std::fmt;
+use std::marker::PhantomData;
+use std::mem;
+use std::os::raw::c_void;
+use std::slice;
+use std::result;
+
+#[derive(Debug,Copy,Clone,PartialEq,Eq)]
+pub struct Error(&'static str);
```

Comment:
I try to use `error_chain` as much as possible but doesn't need to be part of this PR.  Would be cool if you could put this in an `error` module though so that I can eventually use `error_chain` without having to `pub use` this.